utils/getInvoices/tests/getInvoices.discount.test.ts

```diff
@@ -6,135 +6,452 @@ import { ServiceType } from '@utils/constants'
+      const company: Partial<IRealestate> = {
+        discount: undefined,
+      }
+
+      const invoices = getInvoices({
+        company,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
     })
   })
+  describe('props: { service }', () => {
+    it('should NOT load when service = null', () => {
+      const service: Partial<IService> = null
+
+      const invoices = getInvoices({
+        service,
+      })
 
-  it('should load Discount price from payment as sum', () => {
-    const company: Partial<IRealestate> = null
-    const service: Partial<IService> = null
-    const payment: Partial<IPayment> = {
-      invoice: [
-        {
-          type: ServiceType.Discount,
-          price: 0,
-          sum: 100,
-        },
-      ],
-    }
-    const prevPayment: Partial<IPayment> = null
-
-    const invoices = getInvoices({
-      company,
-      service,
-      payment,
-      prevPayment,
-    })
-
-    expect(invoices).toContainEqual({
-      type: ServiceType.Discount,
-      price: payment.invoice[0].sum,
-      sum: payment.invoice[0].sum,
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
+    })
+    it('should NOT load when service = { rentPrice: 1000 }', () => {
+      const service: Partial<IService> = {
+        rentPrice: 1000,
+      }
+
+      const invoices = getInvoices({
+        service,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
+    })
+    it('should NOT load when service = { rentPrice: 0 }', () => {
+      const service: Partial<IService> = {
+        rentPrice: 0,
+      }
+
+      const invoices = getInvoices({
+        service,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
+    })
+    it('should NOT load when service = { rentPrice: NaN }', () => {
+      const service: Partial<IService> = {
+        rentPrice: NaN,
+      }
+
+      const invoices = getInvoices({
+        service,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
+    })
+    it('should NOT load when service = { rentPrice: null }', () => {
+      const service: Partial<IService> = {
+        rentPrice: null,
+      }
+
+      const invoices = getInvoices({
+        service,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
+    })
+    it('should NOT load when service = { rentPrice: undefined }', () => {
+      const service: Partial<IService> = {
+        rentPrice: undefined,
+      }
+
+      const invoices = getInvoices({
+        service,
+      })
+
+      expect(invoices).not.toContainEqual(
+        expect.objectContaining({ type: ServiceType.Discount })
+      )
```

Comment:
Same behavior can be joined to a single test `should NOT load when service = { ... }`